src/main.ts

```diff
@@ -83,11 +83,19 @@ async function run(): Promise<void> {
     // Deploy function
     const deployFunctionResponse = await client.deploy(newFunc);
 
-    if (deployFunctionResponse.response?.httpsTrigger?.url) {
-      // Set Cloud Function URL as output
-      core.setOutput('url', deployFunctionResponse.response.httpsTrigger.url);
+    const resp = deployFunctionResponse.response;
+    if (resp) {
+      if (resp.httpsTrigger?.url) {
+        core.setOutput('url', resp.httpsTrigger.url);
+      } else {
+        core.info('No URL set. Only HttpsTrigger Cloud Functions have URL.');
+      }
+      core.setOutput('name', resp.name);
```

Comment:
Since `name`, `runtime` are inputs and not computer server side, are there any use cases for setting this as op?

Reply:
name is computed server side. Name is `projects/p/locations/l/...` and it's super useful in tests and other automation that wants a full resource URL. `runtime` can also be different. `node` will currently default and change to `node10` (I think)

Reply:
Since we take in an input `name` I think it could be confusing if we return an output with the same name.  Maybe `id` or `self_link` might be a better output name?

Reply:
Switched to `id`